src/OmenThumbnailMapping.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+interface IFixedProductMarketMaker {
+    function balanceOf(address account) external view returns (uint256);
+}
+
+contract OmenThumbnailMapping {
+    // Mapping from market's address to image's IPFS hash
+    mapping(address => string) private marketAddressToIPFSHash;
+    mapping(address => address) private marketAddressToLatestImageChanger;
+
+    function get(address marketAddress) public view returns (string memory) {
+        // Get IPFS hash of thumbnail for the given market.
+        return marketAddressToIPFSHash[marketAddress];
+    }
+
+    function set(address marketAddress, string memory image_hash) public {
+        // Update IPFS hash of thumbnail for the given market.
+        requireThatSenderCanChangeImage(marketAddress);
+        marketAddressToIPFSHash[marketAddress] = image_hash;
+        marketAddressToLatestImageChanger[marketAddress] = msg.sender;
+    }
+
+    function remove(address marketAddress) public {
+        // Remove IPFS hash of thumbnail for the given market.
+        requireThatSenderCanChangeImage(marketAddress);
+        delete marketAddressToIPFSHash[marketAddress];
+        marketAddressToLatestImageChanger[marketAddress] = msg.sender;
+    }
+
+    function requireThatSenderCanChangeImage(
+        address marketAddress
+    ) public view {
```

Comment:
Change this to a modifier -> https://www.freecodecamp.org/news/what-are-solidity-modifiers/ for reusability

Reply:
Cool stuff, used.